src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/TimestamperUtil.java

```diff
@@ -50,48 +52,46 @@ private TimestamperUtil() {
     }
 
     @SuppressWarnings("unchecked")
-    static TimeStampToken addSignerCertificate(
-            TimeStampResponse tsResponse, X509Certificate signerCertificate)
-                    throws Exception {
+    static TimeStampToken addSignerCertificate(TimeStampResponse tsResponse,
+            X509Certificate signerCertificate) throws Exception {
         CMSSignedData cms = tsResponse.getTimeStampToken().toCMSSignedData();
 
-        List<X509Certificate> collection = Arrays.asList(signerCertificate);
+        List<X509Certificate> collection = Collections.singletonList(signerCertificate);
         collection.addAll(cms.getCertificates().getMatches(null));
```

Comment:
The _collection_ is immutable (a singletonlist). AddAll will fail if the argument is not an empty collection.